client.go

```diff
@@ -489,10 +500,21 @@ func (c *Client) httpRequest(method string, path string, data []byte, headers ma
 	var err error
 	for i := 0; i < maxHosts; i++ {
 		reader := bytes.NewReader(data)
-		// get a host from the cluster
-		host := c.cluster.Host()
-		if host == nil {
-			return nil, nil, ErrEmptyCluster
+		var host *URI
+
+		if useCoordinator {
+			// TODO: we shouldn't need to fetch the coordinator every time
+			node, err := c.fetchCoordinatorNode()
```

Comment:
I feel like we should only be fetching the coordinator node if a request fails because it should have been directed to the coordinator, but wasn't.

Reply:
I agree. I thought about adding a `coordinatorNode` field to the `PilosaClient` struct but then I have to add a mutex to prevent races. This is meant to be something that's safe until we find a better solution (like initializing `PilosaClient` with the coordinator node, etc.)

Reply:
ok, I agree we can defer further work on this